domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/display/DisplayGroupAdapterService.java

```diff
@@ -45,74 +53,100 @@ public WizardPageCollection findWizardPagesByCaseTypeId(String caseTypeReference
-                                override.setComplexFieldElementId(reference + "." + eventComplexTypeEntity.getReference());
-                                override.setDisplayContext(eventComplexTypeEntity.getDisplayContext().toString());
-                                override.setOrder(eventComplexTypeEntity.getOrder());
-                                override.setLabel(eventComplexTypeEntity.getLabel());
-                                override.setHintText(eventComplexTypeEntity.getHint());
-                                override.setShowCondition(eventComplexTypeEntity.getShowCondition());
-
-                                wizardPageField.addComplexFieldOverride(override);
-                            });
-
-                            List<String> complexFieldOverrideIds = wizardPageField.getComplexFieldOverrides()
-                                .stream()
-                                .map(WizardPageComplexFieldOverride::getComplexFieldElementId)
-                                .collect(Collectors.toList());
-
-                            List<String> hiddenFieldOverrideToCreate = allSubTypePossibilities.stream()
-                                .filter(e -> e.startsWith(displayGroupCaseFieldEntity.getCaseField().getReference()))
-                                .filter(e -> !e.equals(displayGroupCaseFieldEntity.getCaseField().getReference()))
-                                .filter(e -> !complexFieldOverrideIds.contains(e))
-                                .collect(Collectors.toList());
-
-                            hiddenFieldOverrideToCreate.forEach(hiddenFieldOverride ->
-                                wizardPageField.addComplexFieldOverride(hiddenWizardPageComplexFieldOverride(hiddenFieldOverride)));
-                        }
-                        wizardPage.getWizardPageFields().add(wizardPageField);
-                    });
-                    wizardPageCollection.getWizardPages().add(wizardPage);
-                });
+                wizardPageCollection.getWizardPages().addAll(displayGroupEntityList.stream()
+                    .map(this::createWizardPage)
+                    .collect(Collectors.toList()));
+
                 return wizardPageCollection;
             }
         }
         return null;
     }
 
+    private WizardPage createWizardPage(final DisplayGroupEntity displayGroupEntity) {
+        WizardPage wizardPage = new WizardPage();
+        wizardPage.setId(displayGroupEntity.getReference());
+        wizardPage.setLabel(displayGroupEntity.getLabel());
+        wizardPage.setOrder(displayGroupEntity.getOrder());
+
+        Optional.ofNullable(displayGroupEntity.getWebhookMidEvent()).ifPresent(webhookEntity -> {
+            wizardPage.setCallBackURLMidEvent(webhookEntity.getUrl());
+            wizardPage.setRetriesTimeoutMidEvent(webhookEntity.getTimeouts());
+        });
+
+        List<String> allSubTypePossibilities = createAllSubtypeLeafCombinations(displayGroupEntity);
```

Comment:
any reason to create the variable here and pass it to `createWizardCaseField` rather than creating it inside `createWizardCaseField` itself?

Reply:
Because it is out of displayGroupEntity.getDisplayGroupCaseFields().stream() so it gets computed once.